clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -324,6 +324,15 @@ void cir::AllocaOp::build(::mlir::OpBuilder &odsBuilder,
 // BreakOp
 //===----------------------------------------------------------------------===//
 
+mlir::Operation *cir::BreakOp::getBreakTarget() {
+  // Find the innermost loop or switch parent operation.
+  mlir::Operation *parentOp = getOperation()->getParentOp();
+  while (::mlir::isa_and_present<cir::LoopOpInterface, cir::SwitchOp>(parentOp))
```

Comment:
```suggestion
  while (mlir::isa_and_present<cir::LoopOpInterface, cir::SwitchOp>(parentOp))
```

Reply:
I would prefer separate this list. `BreakableTrait` is probably overkill as I don't see more items to be added to the list, but `bool isBreakableOp(mlir::Operation *op)` predicate would be nicer.

So you have then `while(isBreakableOp(parentOp)) ...`

Reply:
Also it should be probably negated